jgt_common/__init__.py

```diff
@@ -1164,6 +1166,20 @@ def build_classification_rst_string(from_dict, for_module, category_name_mapping
     return result
 
 
+@classify("misc")
+def percent_diff(a, b, precision=2):
+    """Get percentage difference, out to ``precision`` places."""
+    if a == b:
+        return 0
+    elif a == 0 or b == 0:
+        return nan
+    # It is debatable whether this is the correct choice for determining
+    # relative difference. min(), max(), and the mean all have a case to be made for
+    # them, both with and without abs(). See, e.g.:
+    # https://en.wikipedia.org/wiki/Relative_change_and_difference#Definitions
+    return round(abs((a - b) / max(abs(a), abs(b))) * 100, precision)
```

Comment:
To add to this comment, I feel like the `precision` parameter might not really be all that useful. Perhaps adding a choice of scale functions, similar to [this npm package](https://www.npmjs.com/package/math-relative-difference) would be a better choice. Also like that package, perhaps we should return the relative difference directly, and let the user multiply by 100 to get a percentage value.

Reply:
I kinda like that idea, except I don't like the string or function aspect of it.
Probably I would favor just always having a function, provide a bunch of nicely named functions to choose from (vs. having them encoded as strings) with a default to match the code you already have?

Reply:
So, have, say, an enum of a bunch of scale functions and then to call it with one of the others, something like:

```
jgt_common.percent_diff(3, 2, scale_func=mean_abs)
```

This lets users pass in their own oddball scale func if they want.